pages/learn/develop/blocks.md

```diff
@@ -0,0 +1,128 @@
+  connected display
+version: 2.3.2
+type: sw.block
+assets:
+  repository:
+    type: blob.asset
+    data:
+      url: 'https://github.com/balenablocks/browser'
+  logo:
+    type: blob.asset
+    data:
+      url: 'https://raw.githubusercontent.com/balenablocks/browser/master/logo.png'
+data:
+  defaultDeviceType: raspberrypi3
+  supportedDeviceTypes:
+    - raspberrypi4-64
+    - fincm3
+    - raspberrypi3
+    - raspberrypi3-64
+    - raspberrypi400-64
+    - intel-nuc
+    - genericx86-64-ext
+```
+
+Next, navigate to the `Blocks` tab in the sidebar and click the `Create block` button in the balenaCloud dashboard.
+
+<img src="/img/blocks/image.png" width="100%">
+
+Fill the details for your block in the modal, and click `Create new block` button to create your block.
+
+<img src="/img/blocks/create-block.png" width="70%">
+
+Next, we have to [push a new release][balena-push] from your project to your newly created block on balenaCloud. If you are new to this, then follow the [Getting Started][getting-started] guide. After the release has been created, your block is ready. You can manage, modify and iterate further on your block from the balenaCloud dashboard.
+
+## Release management for blocks
+
+To deploy new releases of your block, you can use the [balena-push][balena-push] command using the [balenaCLI](https://github.com/balena-io/balena-cli). The `Releases` tab in the sidebar will show the list of past releases of your block as well as the release that is currently selected as default.
+
+<img src="/img/blocks/release.png" width="80%">
+
+By default, the block set to `track latest`, which means that new releases are immediately tagged as the default release for the block. Expanding this _Default release_ dropdown menu displays all releases for the block, and you can select a specific release to pin the block to. When a particular release is selected, that release will be tagged as the default (or latest) release of your block. In future, releases will not be deployed until the _Default release_ is updated to a newer release or set back to `track latest`.
+
+<img src="/img/blocks/release-mgmt.png" width="100%">
+
+## Making your block public
+
+When a block is created on balenaCloud it will not be publicly visible on balenaHub. When a block is made public it is listed in the [Blocks section][blockhub] of balenaHub, but is not joinable like fleets or forkable like projects. Instead the card for the block allows users to find the source code for the block, and learn how to deploy and use it in their own fleet.
+
+When blocks are private, you can iterate, test and build more functionality before making it available to the world. When you are ready to release, head to the `Settings` tab in the sidebar and click the `Make public` button to toggle the visibility of your block on balenaHub; you’re free to toggle the visibility on and off as necessary at any time.
```

Comment:
As Phil mentioned in FD, there's no `Make public` button. I know you're aware of this but just leaving this comment so we don't forget to update this.

Also we should specify here that a block cannot be made publicly visible unless there is a repo URL provided (even though this won't be a _requirement_ anymore in the future, rather just an option. When that happens we'll need to update the docs again)